op-chain-ops/cmd/celo-migrate/db.go

```diff
@@ -34,35 +38,155 @@ func headerKey(number uint64, hash common.Hash) []byte {
+	}
+
+	newDB, err := openDBWithoutFreezer(newDBPath, true)
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			return false, nil
+		}
+		return false, fmt.Errorf("failed to open new database: %w", err)
+	}
+	defer newDB.Close()
+
+	marker, err := readFullMigrationMarker(newDB)
+	if err != nil {
+		if errors.Is(err, leveldb.ErrNotFound) {
+			return false, nil
+		}
+		return false, fmt.Errorf("failed to read celo migration marker: %w", err)
+	}
+
+	return len(marker) > 0, nil
+}
+
+// TODO(Alec) delete these
+// func checkForPrevFullMigration(newDBPath string) (bool, error) {
+// 	newDB, err := openDB(newDBPath)
+// 	if err != nil {
+// 		return false, fmt.Errorf("failed to open new database: %w", err)
+// 	}
+// 	defer newDB.Close()
+
+// 	// Get body of first non-ancient block and see if it's already been transformed
+// 	numAncients, err := newDB.Ancients()
+// 	if err != nil {
+// 		return false, fmt.Errorf("failed to read numAncients from new database: %w", err)
+// 	}
+// 	hash := rawdb.ReadCanonicalHash(newDB, numAncients)
+// 	bodyData := rawdb.ReadBodyRLP(newDB, hash, numAncients)
+
+// 	log.Info("bodyData: ", bodyData, "numAncients: ", numAncients, "hash: ", hash)
+
+// 	// Try decoding body into celo-blockchain Body structure
+// 	var celoBody struct {
+// 		Transactions   types.Transactions
+// 		Randomness     rlp.RawValue
+// 		EpochSnarkData rlp.RawValue
+// 	}
+// 	if err := rlp.DecodeBytes(bodyData, &celoBody); err != nil {
+// 		// body didn't decode, may have already been transformed in a previous migration
+// 		body := types.Body{}
+// 		if err := rlp.DecodeBytes(bodyData, &body); err == nil {
+// 			// body decodes into our new type, so it's already been transformed in a previous migration
+// 			return true, nil
+// 		}
+// 		return false, fmt.Errorf("failed to RLP decode body: %w", err)
+// 	}
+
+// 	return false, nil
+// }
+
+// func checkForPrevFullMigration(oldDBPath, newDBPath string) (bool, error) {
+// 	// Compare the last modified time of files in both directories
+// 	newFiles, err := os.ReadDir(newDBPath)
+// 	if err != nil {
+// 		return false, fmt.Errorf("failed to read directory: %w", err)
+// 	}
+
+// 	for _, newFile := range newFiles {
+// 		if newFile.Name() != "ancient" {
+// 			newPath := filepath.Join(newDBPath, newFile.Name())
+// 			oldPath := filepath.Join(oldDBPath, newFile.Name())
+
+// 			newFileInfo, err := os.Stat(newPath)
+// 			if err != nil {
+// 				return false, fmt.Errorf("failed to get file info for %s: %w", newPath, err)
+// 			}
+// 			oldFileInfo, err := os.Stat(oldPath)
+// 			if err != nil {
+// 				return false, fmt.Errorf("failed to get file info for %s: %w", oldPath, err)
+// 			}
+
+// 			if newFileInfo.ModTime().After(oldFileInfo.ModTime()) {
+// 				// New file has been modified more recently
+// 				return true, nil
+// 			}
+// 		}
+// 	}
+
+// 	return false, nil
+// }
+
+func resetDbIfNeeded(newDbPath string, measureTime bool) error {
```

Comment:
We don't need this method any more, because the rsync command will delete anything that is not in the source directory. And that means we also don't need to check for a previous full migration.

Reply:
Ahh good point. I'll test that just to make sure

Reply:
It looks like this does indeed work. After testing a bunch of times the migrated directory seems to end up in the same state after running the script repeatedly with the resetDB logic removed.